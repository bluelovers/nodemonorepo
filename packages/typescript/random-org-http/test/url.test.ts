import {url, types} from '../index'
import Param = types.raw.Param
import Radix = types.Base
import Format = types.Format
import Activation = types.Activation

const {integers, sequences, strings} = url.create

const mksmpl =
  <Param, Result>(
    fn: (param: Param) => Result,
    params: ReadonlyArray<Param>
  ) => params.map(
    input => ({
      input,
      output: fn(input)
    })
  )

it('integers', () => {
  const params: Param.Integer[] = [
    {
      base: Radix.binary,
      col: 1,
      min: 1,
      max: 10,
      num: 8,
      format: Format.plain
    },
    {
      base: Radix.hexadecimal,
      col: 3,
      min: -10,
      max: 20,
      num: 7,
      format: Format.html
    }
  ]

  const result = mksmpl(integers, params)

  expect(result).toMatchSnapshot()
})

it('sequences', () => {
  const params: Param.Sequence[] = [
    {
      min: 1,
      max: 10,
      col: 2,
      format: Format.plain
    },
    {
      min: 2,
      max: 5,
      col: 3,
      format: Format.html
    }
  ]

  const result = mksmpl(sequences, params)

  expect(result).toMatchSnapshot()
})

it('strings', () => {
  const params: Param.String[] = [
    {
      num: 1,
      len: 10,
      digits: Activation.on,
      loweralpha: Activation.off,
      upperalpha: Activation.on,
      unique: Activation.off,
      format: Format.plain
    },
    {
      num: 7,
      len: 15,
      digits: Activation.on,
      loweralpha: Activation.on,
      upperalpha: Activation.on,
      unique: Activation.on,
      format: Format.html
    },
    {
      num: 9,
      len: 19,
      digits: Activation.on,
      loweralpha: Activation.off,
      upperalpha: Activation.off,
      unique: Activation.off,
      format: Format.plain
    }
  ]

  const result = mksmpl(strings, params)

  expect(result).toMatchSnapshot()
})
