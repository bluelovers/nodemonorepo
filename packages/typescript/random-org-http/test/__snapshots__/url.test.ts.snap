// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integer 1`] = `
Array [
  Object {
    "input": Object {
      "base": 2,
      "col": 1,
      "format": "plain",
      "max": 10,
      "min": 1,
      "num": 8,
    },
    "output": "https://random.org/integers/?base=2&col=1&min=1&max=10&num=8&format=plain",
  },
  Object {
    "input": Object {
      "base": 16,
      "col": 3,
      "format": "html",
      "max": 20,
      "min": 10,
      "num": 7,
    },
    "output": "https://random.org/integers/?base=16&col=3&min=10&max=20&num=7&format=html",
  },
]
`;

exports[`sequence 1`] = `
Array [
  Object {
    "input": Object {
      "col": 2,
      "format": "plain",
      "max": 10,
      "min": 1,
    },
    "output": "https://random.org/sequences/?min=1&max=10&col=2&format=plain",
  },
  Object {
    "input": Object {
      "col": 3,
      "format": "html",
      "max": 5,
      "min": 2,
    },
    "output": "https://random.org/sequences/?min=2&max=5&col=3&format=html",
  },
]
`;

exports[`string 1`] = `
Array [
  Object {
    "input": Object {
      "digits": "on",
      "format": "plain",
      "len": 10,
      "loweralpha": "off",
      "num": 1,
      "unique": "off",
      "upperalpha": "on",
    },
    "output": "https://random.org/strings/?num=1&len=10&digits=on&loweralpha=off&upperalpha=on&unique=off&format=plain",
  },
  Object {
    "input": Object {
      "digits": "on",
      "format": "html",
      "len": 22,
      "loweralpha": "on",
      "num": 7,
      "unique": "on",
      "upperalpha": "on",
    },
    "output": "https://random.org/strings/?num=7&len=22&digits=on&loweralpha=on&upperalpha=on&unique=on&format=html",
  },
  Object {
    "input": Object {
      "digits": "on",
      "format": "plain",
      "len": 32,
      "loweralpha": "off",
      "num": 9,
      "unique": "off",
      "upperalpha": "off",
    },
    "output": "https://random.org/strings/?num=9&len=32&digits=on&loweralpha=off&upperalpha=off&unique=off&format=plain",
  },
]
`;
